File: medplum-typescript/packages/sdk/src/model/fhir/ObservationDefinition.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { ObservationdefinitionPermitteddatatypeItem } from "./ObservationdefinitionPermitteddatatypeItem";
import { ObservationDefinition_QualifiedInterval } from "./ObservationDefinition_QualifiedInterval";
import { ObservationDefinition_QuantitativeDetails } from "./ObservationDefinition_QuantitativeDetails";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface ObservationDefinition {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    category: CodeableConcept[] | null | undefined;
    code: CodeableConcept;
    identifier: Identifier[] | null | undefined;
    permittedDataType:
        | ObservationdefinitionPermitteddatatypeItem[]
        | null
        | undefined;
    multipleResultsAllowed: boolean | null | undefined;
    method: CodeableConcept | null | undefined;
    preferredReportName: string | null | undefined;
    quantitativeDetails:
        | ObservationDefinition_QuantitativeDetails
        | null
        | undefined;
    qualifiedInterval:
        | ObservationDefinition_QualifiedInterval[]
        | null
        | undefined;
    validCodedValueSet: Reference | null | undefined;
    normalCodedValueSet: Reference | null | undefined;
    abnormalCodedValueSet: Reference | null | undefined;
    criticalCodedValueSet: Reference | null | undefined;
}

File: medplum-java/medplum-model/src/main/java/com/fhir/ObservationDefinition.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.Boolean;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableObservationDefinition.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ObservationDefinition {
  String resourceType();

  Optional<uri> implicitRules();

  Optional<List<ResourceList>> contained();

  Optional<Reference> criticalCodedValueSet();

  Optional<Meta> meta();

  Optional<code> language();

  Optional<CodeableConcept> method();

  Optional<ObservationDefinition_QuantitativeDetails> quantitativeDetails();

  Optional<List<ObservationdefinitionPermitteddatatypeItem>> permittedDataType();

  Optional<Reference> normalCodedValueSet();

  Optional<Boolean> multipleResultsAllowed();

  Optional<id> id();

  Optional<Narrative> text();

  Optional<List<Extension>> extension();

  Optional<Reference> validCodedValueSet();

  Optional<Reference> abnormalCodedValueSet();

  Optional<List<CodeableConcept>> category();

  Optional<String> preferredReportName();

  Optional<List<Identifier>> identifier();

  CodeableConcept code();

  Optional<List<ObservationDefinition_QualifiedInterval>> qualifiedInterval();

  Optional<List<Extension>> modifierExtension();

  static ImmutableObservationDefinition.ResourceTypeBuildStage builder() {
    return ImmutableObservationDefinition.builder();
  }
}
